### PR TITLE
Utils

### PR DESCRIPTION
**Pull request by allanchau**
This PR adds some common files into the common folder and includes a `util` module.

Note the `util` module also acts as a wrapper for the node util module. That means you can do things such as `const { inspect } = require('idearium-lib/util')`

### CODE REVIEW

File: idearium-lib/common/opbeat.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+const includeOpbeat = Object.keys(process.env)
+    .includes('OPBEAT_OVERRIDE');
+const opbeatEnabled = includeOpbeat || !isDevelopment(config.get('env'));
+
+// Setup Opbeat to query the data.
+opbeat.start({
+    active: opbeatEnabled,
+    appId: config.get('opbeatAppId'),
+    captureExceptions: true,
+    ignoreUrls: ['/ping'],
```
Comment (smebberson) [inline idearium-lib/common/opbeat.js]: I think this should be configurable via an ENV/config variable.

File: idearium-lib/index.js [hunk 2]
```diff
@@ -5,7 +5,8 @@ module.exports = {
     // Packages here.
     mq: require('./lib/mq'),
     logs: require('./lib/logs'),
-    utils: require('./lib/utils'),
+    util: require('./lib/util'),
+    utils: require('./lib/utils'), // This should eventually be merged into the util module above.
```
Comment (smebberson) [inline idearium-lib/index.js]: I think we should do this now. Introducing `util` while having `utils` is going to be very confusing. What we should do is:

- keep the function in `utils` but wrap it with https://nodejs.org/api/util.html#util_util_deprecate_function_string
- copy the one function across to `util`.

File: idearium-lib/lib/util/index.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+const find = require('./find');
+const has = require('./has');
+const is = require('./is');
+const to = require('./to');
+// Destructure to remove deprecated and conflicting apis.
+const { debuglog, deprecate, format, inherits, inspect } = require('util');
+
+// Merge all the functions and the node built in util library.
+const utilLib = Object.assign(
```
Comment (smebberson) [inline idearium-lib/lib/util/index.js]: I don't think you need the `utilLib` variable. Just export the result directly. Unnecessary variables is something that Eric Elliot writes about too.

File: idearium-lib/lib/util/is.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ * @return {Boolean} True if the object is really an object.
+ */
+const isObject = obj => !isEqual(obj, null) && !isArray(obj) && obj === Object(obj);
+
+/**
+ * Determine if the environment is production.
+ * @param {String} str String to compare.
+ * @return {Boolean} True if the environment is production.
+ */
+const isProduction = str => str.toLowerCase() === 'production';
```
Comment (smebberson) [inline idearium-lib/lib/util/is.js]: Can we have `isBeta`, `isStaging` too?

**Comment by allanchau** [comment]
@smebberson All the changes have been made, not sure why the builds are throwing errors though. `[SYSTEM] Error: Failed to run composition: Running unit tests; caused by E
rror: Composition candidate lib failed with exit code: 1`

**Pull request merged**